app/views/questions/index.html.erb

```diff
@@ -64,77 +88,53 @@
           <tr>
-            <p class="draggable draggable-text" draggable="true"><td><%= question.title %></td><span class="close">&times;</span></p>
+            <p class="list-group-item draggable-question draggable-text" draggable="true"><td><%= question.title %></td></p>
           </tr>
         <% end %></div>
       </div>
     </div>
 
     <script>
-        const draggables = document.querySelectorAll('.draggable')
-        const containers = document.querySelectorAll('.container')
-
-        for (let i = 0; i < draggables.length; i ++) {
-          const draggable = draggables[i];
-
-          draggable.addEventListener('dragstart', function () {
-            draggedItem = draggable;
-            setTimeout(function () {
-              draggable.style.display = 'none';
-            }, 0)
-          });
-
-          draggable.addEventListener('dragend', function () {
-            draggedItem = draggable;
-            setTimeout(function () {
-              draggable.style.display = 'block';
-              draggedItem = null;
-            }, 0)
-          });
-
-          const close = draggable.getElementsByClassName("close")
-          close[0].style.display = "none";
-          close[0].onclick = function() {
-            containers[0].removeChild(draggable);
-          }
-
-          for (let j = 0; j < containers.length; j ++) {
-            const container = containers[j];
-
-            container.addEventListener('dragover', function (e) {
-              e.preventDefault();
-            });
-            container.addEventListener('dragenter', function (e) {
-              e.preventDefault();
-            });
-            container.addEventListener('drop', function (e) {
-              const temp = draggedItem.getElementsByClassName("close");
-              if (temp[0].style.display == "block") {
-                temp[0].style.display = "none";
-              } else {
-                temp[0].style.display = "block";
-              }
-              this.append(draggedItem);
-            });
-          }
-        }
-    </script>
+    $(".draggable-question").draggable(
+      {helper: "clone"}
+    );
 
+    $(".mcontainer").droppable({
+      accept: ".draggable-question",
+      drop: function(event,ui){
+        var tag = document.createElement("p");
+        tag.append($(ui.helper).html());
+        tag.classList.add("list-group-item");
+        tag.classList.add("draggable-question");
+        tag.classList.add("draggable-text");
+        var closetag = document.createElement("button");
+        var spantag = document.createElement("span");
+        spantag.innerHTML = "&times;";
+        spantag.setAttribute('aria-hidden', 'true');
+        closetag.setAttribute('aria-label', 'Close');
+        closetag.classList.add("close");
+        closetag.type = "button";
+        closetag.appendChild(spantag);
+        tag.append(closetag);
+        $(this).append(tag);    
+      }
```

Comment:
This whole section seems like it has the potential to be done in a cleaner way. Maybe check the API to see if there's a function that can add attributes to classList in one line